test/unit/Github/Api/V3/CreateMilestoneTest.php

```diff
@@ -0,0 +1,168 @@
+use Laminas\AutomaticReleases\Github\Api\V3\CreateMilestoneThroughApiCall;
+use Laminas\AutomaticReleases\Github\Value\RepositoryName;
+use Laminas\Diactoros\Request;
+use Laminas\Diactoros\Response;
+use PHPUnit\Framework\MockObject\MockObject;
+use PHPUnit\Framework\TestCase;
+use Psr\Http\Client\ClientInterface;
+use Psr\Http\Message\RequestFactoryInterface;
+use Psr\Http\Message\RequestInterface;
+use Psr\Log\LoggerInterface;
+use Webmozart\Assert\Assert;
+
+use function uniqid;
+
+class CreateMilestoneTest extends TestCase
+{
+    /** @var ClientInterface&MockObject */
+    private $httpClient;
+    /** @var RequestFactoryInterface&MockObject */
+    private $messageFactory;
+    /** @var LoggerInterface&MockObject */
+    private $logger;
+    /** @psalm-var non-empty-string */
+    private string $apiToken;
+    private CreateMilestoneThroughApiCall $createMilestone;
+
+    protected function setUp(): void
+    {
+        parent::setUp();
+
+        $this->httpClient     = $this->createMock(ClientInterface::class);
+        $this->messageFactory = $this->createMock(RequestFactoryInterface::class);
+        $this->logger         = $this->createMock(LoggerInterface::class);
+        $apiToken             = uniqid('apiToken', true);
+
+        Assert::notEmpty($apiToken);
+
+        $this->apiToken        = $apiToken;
+        $this->createMilestone = new CreateMilestoneThroughApiCall(
+            $this->messageFactory,
+            $this->httpClient,
+            $this->apiToken,
+            $this->logger
+        );
+    }
+
+    public function testSuccessfulRequest(): void
+    {
+        $this->messageFactory
+            ->expects(self::any())
+            ->method('createRequest')
+            ->with('POST', 'https://api.github.com/repos/foo/bar/milestones')
+            ->willReturn(new Request('https://the-domain.com/the-path'));
+
+        $validResponse = (new Response())->withStatus(201);
+
+        $validResponse->getBody()->write(
+            <<<'JSON'
+            {
+                "html_url": "http://another-domain.com/the-pr"
+            }
+            JSON
+        );
+
+        $this->httpClient
+            ->expects(self::once())
+            ->method('sendRequest')
+            ->with(self::callback(function (RequestInterface $request): bool {
+                self::assertSame(
+                    [
+                        'Host'          => ['the-domain.com'],
+                        'Content-Type'  => ['application/json'],
+                        'User-Agent'    => ['Ocramius\'s minimal API V3 client'],
+                        'Authorization' => ['token ' . $this->apiToken],
+                    ],
+                    $request->getHeaders()
+                );
+
+                self::assertJsonStringEqualsJsonString(
+                    <<<'JSON'
+                    {
+                        "title": "1.2.3"
+                    }
+                    JSON
+                    ,
```

Comment:
```suggestion
                    JSON,
```

PHP 7.3 allows following the marker with other symbols, too. :smile: